lib/fs.js

```diff
@@ -211,42 +215,59 @@ Object.defineProperties(Stats.prototype, {
     configurable: true,
     enumerable: true,
     get() {
-      return this._atime !== undefined ?
+      return this._atime !== null ?
           this._atime :
-          (this._atime = new Date(this._atim_msec + 0.5));
+          (this._atime = new Date(this.atimeMs + 0.5));
     },
-    set(value) { return this._atime = value; }
+    set(value) {
+      this.atimeMs = Number(value);
+      return this._atime = value;
+    }
   },
   mtime: {
     configurable: true,
     enumerable: true,
     get() {
-      return this._mtime !== undefined ?
+      return this._mtime !== null ?
           this._mtime :
-          (this._mtime = new Date(this._mtim_msec + 0.5));
+          (this._mtime = new Date(this.mtimeMs + 0.5));
     },
-    set(value) { return this._mtime = value; }
+    set(value) {
+      this.mtimeMs = Number(value);
+      return this._mtime = value;
+    }
   },
   ctime: {
     configurable: true,
     enumerable: true,
     get() {
-      return this._ctime !== undefined ?
+      return this._ctime !== null ?
         this._ctime :
-        (this._ctime = new Date(this._ctim_msec + 0.5));
+        (this._ctime = new Date(this.ctimeMs + 0.5));
     },
-    set(value) { return this._ctime = value; }
+    set(value) {
+      this.ctimeMs = Number(value);
+      return this._ctime = value;
+    }
   },
   birthtime: {
     configurable: true,
     enumerable: true,
     get() {
-      return this._birthtime !== undefined ?
+      return this._birthtime !== null ?
         this._birthtime :
-        (this._birthtime = new Date(this._birthtim_msec + 0.5));
+        (this._birthtime = new Date(this.birthtimeMs + 0.5));
     },
-    set(value) { return this._birthtime = value; }
+    set(value) {
+      this.birthtimeMs = Number(value);
+      return this._birthtime = value;
+    }
   },
+  [util.inspect.custom]: {
+    configurable: true,
+    enumerable: false,
+    value: function() {return this.toJSON();}
```

Comment:
It's doesn't, but Ack.
I wanted to ask about the general approach of adding a [util.inspect.custom], is it Ok?